### PR TITLE
Texture operations in vertex shader sample

### PR DESCRIPTION
**Pull request by trungtle**
Implemented texture operation inside a vertex shader for displace mapping.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Where did `heightmap.jpg` and `plane.obj` come from?  Did you create them?

File: samples/texture_vertex.html [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        void main()
+        {
+            vec2 sampleCoord = fract(st.xy);
+            float level = textureLevel(diffuse, st);
+
+            // Compute LOD using gradient
+            color = textureLod(diffuse, st, level);
+
+            // Compute flat normal using gradient
+            vec3 fdx = vec3(dFdx(vPosition.x), dFdx(vPosition.y), dFdx(vPosition.z));
```
Comment (pjcozzi) [inline samples/texture_vertex.html]: I _think_ this can be `dFdx(vPosition)`.

See https://www.opengl.org/sdk/docs/man/html/dFdx.xhtml

**Comment by pjcozzi** [comment]
What is WebGL 2 specific about this sample?  Just that VTF is guaranteed compared to WebGL 1 where the number of texture units in the vertex shader can be zero?

**Comment by trungtle** [comment]
I made the plane.obj from maya, and the heightmap.jpg was from an unclear source so I just pushed a new one I made from maya also (maybe I could just use the noise3d utility to make one).

This example isn't specific for WebGL2.

**Comment by pjcozzi** [comment]
Thanks.

**Pull request merged**